gpkit/nomial_data.py

```diff
@@ -43,15 +44,65 @@ def __init__(self, exps=None, cs=None, nomials=None, simplify=True):
         else:
             self.units = None
 
+        self._hashvalue = None
+
     def __hash__(self):
-        if not hasattr(self, "_hash"):
+        if self._hashvalue is None:
             # confirm lengths before calling zip
             assert len(self.exps) == len(self.cs)
-            self._hash = hash(tuple(zip(self.exps, tuple(self.cs))))
-        return self._hash
+            self._hashvalue = hash(tuple(zip(self.exps, self.cs)))
+        return self._hashvalue
 
     def __repr__(self):
-        return "gpkit.%s(%s)" % (self.__class__.__name__, str(self._hash))
+        return "gpkit.%s(%s)" % (self.__class__.__name__, hash(self))
+
+    def _get_varkey(self, var):
```

Comment:
Yeah, it's used in `NomialData.diff`, and it's intended to be used more (and perhaps become public) as we refactor coercion to VarKey.